pkg/filter/driftignore.go

```diff
@@ -39,175 +37,52 @@ func (r *DriftIgnore) readIgnoreFile() error {
 	for lineNumber := 1; scanner.Scan(); lineNumber++ {
 		line := scanner.Text()
-		if line == "" || strings.HasPrefix(line, "#") {
-			logrus.WithFields(logrus.Fields{
-				"line": line,
-			}).Debug("Skipped comment or empty line")
-			continue
-		}
-		typeVal := readDriftIgnoreLine(line)
-		nbArgs := len(typeVal)
-		if nbArgs < 2 {
-			logrus.WithFields(logrus.Fields{
-				"line":    strconv.Itoa(lineNumber),
-				"content": line,
-			}).Warnf("unable to parse line, invalid length, got %d expected >= 2", nbArgs)
-			continue
-		}
-		res := strings.Join(typeVal[0:2], ".")
-		if nbArgs == 2 { // We want to ignore a resource (type.id)
-			logrus.WithFields(logrus.Fields{
-				"type": typeVal[0],
-				"id":   typeVal[1],
-			}).Debug("Found ignore resource rule in .driftignore")
-			resExclusionTypeList := r.resExclusionList
-			if strings.Contains(res, "*") {
-				resExclusionTypeList = r.resExclusionWildcardList
-			}
-			resExclusionTypeList[res] = struct{}{}
-			continue
-		}
-		// Here we want to ignore a drift (type.id.path.to.field)
-		ignoreSublist, exists := r.driftExclusionList[res]
-		if !exists {
-			ignoreSublist = make([]string, 0, 1)
+
+		if len(strings.ReplaceAll(line, " ", "")) <= 0 {
+			continue // empty
 		}
-		path := strings.Join(typeVal[2:], ".")
 
-		logrus.WithFields(logrus.Fields{
-			"type": typeVal[0],
-			"id":   typeVal[1],
-			"path": path,
-		}).Debug("Found ignore resource field rule in .driftignore")
+		if strings.HasPrefix(line, "#") {
+			continue // this is a comment
+		}
+		line = strings.ReplaceAll(line, "/", separator)
 
-		ignoreSublist = append(ignoreSublist, path)
-		r.driftExclusionList[res] = ignoreSublist
+		lines = append(lines, gitignore.ParsePattern(line, nil))
+		if !strings.HasSuffix(line, "*") {
+			line := fmt.Sprintf("%s.*", line)
+			lines = append(lines, gitignore.ParsePattern(line, nil))
+		}
 	}
 
 	if err := scanner.Err(); err != nil {
 		return err
 	}
 
+	r.matcher = gitignore.NewMatcher(lines)
+
 	return nil
 }
 
 func (r *DriftIgnore) IsResourceIgnored(res resource.Resource) bool {
 	strRes := fmt.Sprintf("%s.%s", res.TerraformType(), res.TerraformId())
 
-	if _, isExclusionRule := r.resExclusionList[strRes]; isExclusionRule {
-		return true
-	}
-	for resExclusion := range r.resExclusionWildcardList {
-		if wildcardMatchChecker(strRes, resExclusion) {
-			return true
-		}
-	}
-	return false
-}
-
-func (r *DriftIgnore) IsFieldIgnored(res resource.Resource, path []string) bool {
-	exclusionRules, isExclusionRule := r.driftExclusionList[fmt.Sprintf("%s.%s", res.TerraformType(), res.TerraformId())]
-	exclusionWildcardRules, isExclusionWildcardRule := r.driftExclusionList[fmt.Sprintf("%s.*", res.TerraformType())]
-
-	if !isExclusionRule && !isExclusionWildcardRule {
-		return false
-	}
-
-	if !isExclusionRule {
-		exclusionRules = exclusionWildcardRules
-	}
-
-	if r.isExcluded(exclusionRules, path) {
-		return true
-	}
-
-	return false
+	match := r.match(strRes)
+	return match
```

Comment:
`return r.match(strRes)`

Reply:
yeah sorry I forgot to clean after debuging :D